src/core/server/plugins/discovery/plugins_discovery.test.ts

```diff
@@ -17,251 +17,376 @@
+    const { plugin$, error$ } = discover(new PluginsConfig(pluginConfig, env), coreContext);
+
+    mockFs(
+      {
+        [KIBANA_ROOT]: {
+          src: {
+            plugins: {
+              plugin_a: Plugins.invalid(),
+              plugin_b: Plugins.incomplete(),
+              plugin_c: Plugins.incompatible(),
+              plugin_d: Plugins.missingManifest(),
+            },
+          },
+        },
+      },
+      { createCwd: false }
+    );
+
+    const plugins = await plugin$.pipe(toArray()).toPromise();
+    expect(plugins).toHaveLength(0);
 
-test('properly iterates through plugin search locations', async () => {
-  mockPackage.raw = {
-    branch: 'master',
-    version: '1.2.3',
-    build: {
-      distributable: true,
-      number: 1,
-      sha: '',
-    },
-  };
-
-  const env = Env.createDefault(
-    getEnvOptions({
-      cliArgs: { envName: 'development' },
-    })
-  );
-  const configService = new ConfigService(
-    rawConfigServiceMock.create({ rawConfig: { plugins: { paths: [TEST_EXTRA_PLUGIN_PATH] } } }),
-    env,
-    logger
-  );
-  await configService.setSchema(config.path, config.schema);
-
-  const rawConfig = await configService
-    .atPath<PluginsConfigType>('plugins')
-    .pipe(first())
-    .toPromise();
-  const { plugin$, error$ } = discover(new PluginsConfig(rawConfig, env), {
-    coreId: Symbol(),
-    configService,
-    env,
-    logger,
+    const errors = await error$
+      .pipe(
+        map((error) => error.toString()),
+        toArray()
+      )
+      .toPromise();
+
+    expect(errors).toEqual(
+      expect.arrayContaining([
+        `Error: Unexpected token o in JSON at position 1 (invalid-manifest, ${manifestPath(
+          'plugin_a'
+        )})`,
+        `Error: Plugin manifest must contain an "id" property. (invalid-manifest, ${manifestPath(
+          'plugin_b'
+        )})`,
+        `Error: Plugin "plugin" is only compatible with Kibana version "1", but used Kibana version is "1.2.3". (incompatible-version, ${manifestPath(
+          'plugin_c'
+        )})`,
+      ])
+    );
   });
 
-  const plugins = await plugin$.pipe(toArray()).toPromise();
-  expect(plugins).toHaveLength(4);
-
-  for (const path of [
-    resolve(TEST_PLUGIN_SEARCH_PATHS.nonEmptySrcPlugins, '1'),
-    resolve(TEST_PLUGIN_SEARCH_PATHS.nonEmptySrcPlugins, '3'),
-    resolve(TEST_PLUGIN_SEARCH_PATHS.nonEmptySrcPlugins, '6'),
-    TEST_EXTRA_PLUGIN_PATH,
-  ]) {
-    const discoveredPlugin = plugins.find((plugin) => plugin.path === path)!;
-    expect(discoveredPlugin).toBeInstanceOf(PluginWrapper);
-    expect(discoveredPlugin.configPath).toEqual(['core', 'config']);
-    expect(discoveredPlugin.requiredPlugins).toEqual(['a', 'b']);
-    expect(discoveredPlugin.optionalPlugins).toEqual(['c', 'd']);
-  }
-
-  await expect(
-    error$
+  it('return errors when the plugin search path is not accessible', async () => {
```

Comment:
Should we have a similar test for if the `kibana.json` file is inaccessible?

Reply:
Added. However even in `000` mode, the `stat` operation returns in success (which is normal). The error is therefor catched a little late, in `parseManifest`. It's probably still acceptable (and I don't really want to perform a try-read just to catch the error sooner)